extension/sql/timescale-prometheus.sql

```diff
@@ -1,5 +1,27 @@
 
-SET LOCAL search_path TO DEFAULT;
+-- Set the search path to one that will find all the defitions provided by the
+-- connector. Since the connector can change the schemas it stores things
+-- in we cannot just hardcode the searchpath, instead we switch the search path
+-- to one procided by the connector. If there was no search path provided, we
+-- set the search path to the default, if the user is a superuser, this should
+-- be the search path the connector set, though this does not work if the
+-- extension is installed by pgextwlist. We use timescale_prometheus.search_path
+-- as the guc as it is unlikely to be used by anything else, we cannot use a guc
+-- in the timescale_prometheus_extra namespace, as this will cause the
+-- CREATE EXTENSION to fail.
+DO $$
+    DECLARE
+        val text;
+    BEGIN
+        val := current_setting('timescale_prometheus.search_path');
```

Comment:
I actually think we want this recorded in the _prom_catalog.default table (which already exists) during connector install and read here instead of passed in as a guc. That way the install only requires the connect installed its sql stuff and removing the extra set-this-guc appropriately during extension install step.

Reply:
that would mean that the extension needs to know the `_prom_catalog` schema, and the user couldn't rename it